File: replace.py
import re

with open('snippets/whatsapp-otp-modal.liquid', 'r', encoding='utf-8') as f:
    content = f.read()

target1 = """          <!-- Store Credit / Wallet Section (shown only if store_credit_enabled) -->
          <div id="wa-wallet-section" style="display:none; margin-bottom:12px;">
             <div class="wa-payment-option wa-pay-opt" id="wa-wallet-card" style="margin-top:0; padding:10px 14px; cursor: pointer; transition: 0.2s; border:1.5px dashed #cbd5e1; background:#f8fafc;" onclick="waToggleWalletCredit()">
                <div style="display:flex; width:100%; align-items:center; gap:12px;">
                   <div class="wa-checkbox-btn" id="wa-wallet-checkbox" style="width:18px; height:18px; border-radius:4px; border:2px solid #94a3b8; display:flex; align-items:center; justify-content:center; transition:0.2s;">
                      <svg id="wa-wallet-check-icon" width="12" height="12" viewBox="0 0 24 24" fill="none" stroke="white" stroke-width="3" style="display:none;"><polyline points="20 6 9 17 4 12"></polyline></svg>
                   </div>
                   <div style="flex:1; display:flex; justify-content:space-between; align-items:center;">
                      <div style="display:flex; gap:10px; align-items:center;">
                         <div class="wa-icon-circ" style="background:#e0e7ff; width:28px; height:28px;">
                            <svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="#4f46e5" stroke-width="2"><path d="M21 12V7H5a2 2 0 0 1 2-2h14v4"/><path d="M3 5v14a2 2 0 0 0 2 2h16v-5H5a2 2 0 0 1-2-2z"/></svg>
                         </div>
                         <div>
                           <div style="font-weight:700; color:#1e293b; font-size:14px;">Store Credit Wallet</div>
                           <div style="font-size:11px; color:#64748b; margin-top:2px;">Available: <strong id="wa-wallet-balance-display">₹0</strong></div>
                         </div>
                      </div>
                      <div id="wa-wallet-applied-row" style="display:none; font-size:12px; font-weight:700; color:#4f46e5;">
                         <span id="wa-wallet-applied-amt">-₹0</span>
                      </div>
                   </div>
                </div>
                <div id="wa-wallet-usable-text" style="display:none; font-size:11px; color:#64748b; margin-top:6px; margin-left:30px;">Max usable: ₹0</div>
             </div>
          </div>"""

rep1 = """          <!-- Store Credit / Wallet Section (shown only if store_credit_enabled) -->
          <div id="wa-wallet-section" style="display:none; margin-bottom:12px;">
             <div class="wa-pay-opt" id="wa-wallet-card" style="margin-top:0; padding:12px 14px; cursor: pointer; transition: 0.2s; border:1px solid #e2e8f0; border-radius:10px; background:#f8fafc; display:flex; align-items:center; justify-content:space-between;" onclick="waToggleWalletCredit()">
                
                <div style="display:flex; align-items:center; gap:10px; flex:1;">
                   <div style="background:#e0e7ff; width:28px; height:28px; border-radius:50%; display:flex; align-items:center; justify-content:center; flex-shrink:0;">
                      <svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="#4f46e5" stroke-width="2.5"><rect x="2" y="6" width="20" height="12" rx="2"/><path d="M12 12h.01"/><path d="M17 12h.01"/><path d="M7 12h.01"/></svg>
                   </div>
                   <div style="display:flex; flex-direction:column;">
                     <div style="display:flex; align-items:center; gap:6px;">
                       <span style="font-weight:700; color:#0f172a; font-size:13px;">Use Store Credit</span>
                       <div id="wa-wallet-applied-row" style="display:none; font-size:10px; font-weight:800; color:#4f46e5; background:#eff6ff; padding:2px 6px; border-radius:12px;">
                          <span id="wa-wallet-applied-amt">-₹0</span>
                       </div>
                     </div>
                     <span style="font-size:11px; color:#64748b; font-weight:500;">Balance: <strong id="wa-wallet-balance-display" style="color:#0f172a;">₹0</strong></span>
                     <div id="wa-wallet-usable-text" style="display:none;"></div>
                   </div>
                </div>

                <!-- Toggle Switch Checkbox replacement -->
                <div id="wa-wallet-checkbox" style="position:relative; width:34px; height:20px; background:#cbd5e1; border-radius:20px; transition:0.3s; display:flex; align-items:center; padding:2px; flex-shrink:0; border:none;">
                   <div id="wa-wallet-check-icon" style="width:16px; height:16px; background:#fff; border-radius:50%; box-shadow:0 1px 3px rgba(0,0,0,0.15); transition:0.3s; transform:translateX(0); display:block;"></div>
                </div>
                
             </div>
          </div>"""

content = content.replace(target1, rep1)

target2 = """    if (waWalletApplied) {
      waWalletAppliedAmt = Math.round(computeWalletUsable() * 100) / 100;
      if (card) { card.style.borderColor = 'var(--wa-primary, #0f172a)'; card.style.background = '#f0f9ff'; }
      if (checkbox) { checkbox.style.borderColor = 'var(--wa-primary, #0f172a)'; checkbox.style.background = 'var(--wa-primary, #0f172a)'; }
      if (checkIcon) checkIcon.style.display = 'block';
      if (appliedRow) { appliedRow.style.display = 'flex'; appliedRow.style.background = '#eff6ff'; }
      if (appliedAmt) appliedAmt.innerText = `-₹${waWalletAppliedAmt.toFixed(2)}`;
    } else {
      if (card) { card.style.borderColor = '#cbd5e1'; card.style.background = '#f8fafc'; }
      if (checkbox) { checkbox.style.borderColor = '#94a3b8'; checkbox.style.background = 'transparent'; }
      if (checkIcon) checkIcon.style.display = 'none';
      if (appliedRow) appliedRow.style.display = 'none';
    }"""

rep2 = """    if (waWalletApplied) {
      waWalletAppliedAmt = Math.round(computeWalletUsable() * 100) / 100;
      if (card) { card.style.borderColor = 'var(--wa-primary, #0f172a)'; card.style.background = '#f0f9ff'; }
      if (checkbox) { checkbox.style.background = 'var(--wa-primary, #0f172a)'; }
      if (checkIcon) { checkIcon.style.transform = 'translateX(14px)'; }
      if (appliedRow) { appliedRow.style.display = 'flex'; }
      if (appliedAmt) appliedAmt.innerText = `-₹${waWalletAppliedAmt.toFixed(2)}`;
    } else {
      if (card) { card.style.borderColor = '#e2e8f0'; card.style.background = '#f8fafc'; }
      if (checkbox) { checkbox.style.background = '#cbd5e1'; }
      if (checkIcon) { checkIcon.style.transform = 'translateX(0)'; }
      if (appliedRow) appliedRow.style.display = 'none';
    }"""
content = content.replace(target2, rep2)
with open('snippets/whatsapp-otp-modal.liquid', 'w', encoding='utf-8') as f:
    f.write(content)

File: replace.js
const fs = require('fs');
let content = fs.readFileSync('snippets/whatsapp-otp-modal.liquid', 'utf-8');

function replaceRegex(content, regex, replacement) {
    if (regex.test(content)) {
        return content.replace(regex, replacement);
    }
    console.log("Failed to match: " + regex);
    return content;
}

// Fix 1: Visa Icon URL
content = content.replace(
    /https:\/\/upload\.wikimedia\.org\/wikipedia\/commons\/a\/a4\/Visa_Logo_2012_color\.svg/g,
    'https://cdn.shopify.com/s/assets/payment_icons/visa-319d545c6fd255c9aad5eeaad21fd6f7f7b4fdbdb1a35ce83b89cca12a187f00.svg'
);

// Fix 2: Trust Badge Padding
content = content.replace(
    /<div class="wa-trust-badge-container" id="wa-mid-trust-badges" style="display:flex; justify-content:space-between; align-items:center; padding:16px; border:1px solid #e2e8f0; border-radius:12px; margin-bottom:24px; background:#fff; gap:4px;">/,
    '<div class="wa-trust-badge-container" id="wa-mid-trust-badges" style="display:flex; justify-content:space-between; align-items:center; padding:12px 6px; border:1px solid #e2e8f0; border-radius:12px; margin-bottom:24px; background:#fff; gap:2px;">'
);

// Fix 3: Lock icon text wrapper (Step 1)
content = content.replace(
    /<div style="text-align:center; font-size:11px; color:#475569; font-weight:600; display:flex; justify-content:center; align-items:center; gap:6px;">\s*<svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2"><rect x="3" y="11" width="18" height="11" rx="2" ry="2"\/><path d="M7 11V7a5 5 0 0 1 10 0v4"\/><\/svg>\s*Your security is our priority\. We never share your details\.\s*<\/div>/,
    `<div style="text-align:center; font-size:11px; color:#475569; font-weight:600; display:flex; justify-content:center; align-items:center; gap:6px;">
            <svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" style="flex-shrink:0;"><rect x="3" y="11" width="18" height="11" rx="2" ry="2"/><path d="M7 11V7a5 5 0 0 1 10 0v4"/></svg>
            <span>Your security is our priority. We never share your details.</span>
          </div>`
);

// Fix 4: sendWaOtp button innerHTML reset
let sendWaOtpRegex = /async function sendWaOtp\(\) \{[\s\S]*?btn\.innerHTML = 'Sending OTP\.\.\.';[\s\S]*?finally \{[\s\S]*?btn\.innerHTML = 'Continue[^']*';\s*\}\s*\}/;
let sendWaOtpHit = sendWaOtpRegex.exec(content);
if (sendWaOtpHit) {
    let matchStr = sendWaOtpHit[0];
    matchStr = matchStr.replace("const btn = document.getElementById('wa-send-btn');\n    btn.disabled = true; btn.innerHTML = 'Sending OTP...';", 
        "const btn = document.getElementById('wa-send-btn');\n    const originalBtnHTML = btn.innerHTML;\n    btn.disabled = true; btn.innerHTML = '<span style=\"color:#fff; font-size:16px; font-weight:700;\">Sending OTP...</span>';");
    matchStr = matchStr.replace(/btn\.innerHTML = 'Continue[^']*';/, "btn.innerHTML = originalBtnHTML;");
    content = content.replace(sendWaOtpHit[0], matchStr);
} else {
    console.log("Failed to match sendWaOtp logic");
}

// Fix 5: verifyWaOtp button innerHTML reset
let verifyWaOtpRegex = /async function verifyWaOtp\(\) \{[\s\S]*?btn\.innerHTML = 'Verifying\.\.\.';[\s\S]*?finally \{[\s\S]*?btn\.innerHTML = 'Verify & Continue[^']*';\s*\}\s*\}/;
let verifyWaOtpHit = verifyWaOtpRegex.exec(content);
if (verifyWaOtpHit) {
    let matchStr = verifyWaOtpHit[0];
    matchStr = matchStr.replace("const btn = document.getElementById('wa-verify-btn');\n    if (btn.disabled) return; // Prevent double-submit\n    const otp = waGetOtp();\n    if (otp.length < 4) return;\n    const errEl = document.getElementById('wa-otp-error');\n    btn.disabled = true;\n    btn.innerHTML = 'Verifying...';", 
        "const btn = document.getElementById('wa-verify-btn');\n    if (btn.disabled) return; // Prevent double-submit\n    const otp = waGetOtp();\n    if (otp.length < 4) return;\n    const errEl = document.getElementById('wa-otp-error');\n    const originalBtnHTML = btn.innerHTML;\n    btn.disabled = true;\n    btn.innerHTML = '<span style=\"color:#fff; font-size:16px; font-weight:700;\">Verifying...</span>';");
    matchStr = matchStr.replace(/btn\.innerHTML = 'Verify & Continue[^']*';/, "btn.innerHTML = originalBtnHTML;");
    content = content.replace(verifyWaOtpHit[0], matchStr);
} else {
    console.log("Failed to match verifyWaOtp logic");
}

// Fix 6: waPayNow hardcoded reset
// In the catch blocks inside waPayNow, there are hardcoded resets.
// Let's just blindly replace them if they exist.
content = content.replace(/btn\.innerHTML = '<span id="wa-pay-btn-text">Pay Now<\/span> <svg width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2\.5"><path d="M5 12h14M12 5l7 7-7 7"\/><\/svg>';/g, "btn.innerHTML = originalBtnHTML;");

fs.writeFileSync('snippets/whatsapp-otp-modal.liquid', content, 'utf-8');
console.log('Replaced successfully');
